class.jetpack.php

```diff
@@ -3872,6 +3876,31 @@ function disconnect_survey_notice() {
 		<?php
 	}
 
+	/**
+	 * Checks if the user have started a register flow from calypso in the last day, and if so, redirect them there
+	 */
+	function check_jetpack_connect_redirection() {
+		$user = $this->get_connected_user_data();
+		$parsed_site = parse_url( get_site_url() );
+
+		if ( isset( $user[ 'jetpack_connect' ] ) && is_array( $user[ 'jetpack_connect' ] ) ) {
+			foreach ($user[ 'jetpack_connect' ] as $jetpack_connect_request ) {
+				$parsed_jetpack_connect_site = $jetpack_connect_request[ 'site_url' ];
+
+				if ( $parsed_jetpack_connect_site->domain == $parsed_site->domain
+					&& $parsed_jetpack_connect_site->path == $parsed_site->path
+					&& ( time() - $this->JETPACK_CONNECT_TIMEOUT ) < strtotime( $jetpack_connect_request[ 'date' ] ) ) {
+					// the user started a flow from calypso registering this site url in the last 24 hours
+					$this->message .= __( '<div> Let us take you back to WordPress.com.</div>' );
+					echo '<script> setTimeout( function() { window.location = "' . $this->JETPACK_CONNECT_FLOW_URL . $this->build_raw_urls( get_site_url() ) . '" }, 1000 ); </script>';
```

Comment:
I see. PHP's `header()` redirect allows a delay with `[refresh](http://php.net/manual/en/function.header.php#97114)`, but it's not supported in core's `wp_redirect()`... so either ways would look weird.